server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -1321,7 +1321,7 @@ func streamRawInvocationEvents(env environment.Env, ctx context.Context, streamI
 // in memory.
 func LookupInvocation(env environment.Env, ctx context.Context, iid string) (*inpb.Invocation, error) {
 	var events []*inpb.InvocationEvent
-	inv, err := LookupInvocationWithCallback(ctx, env, iid, func(event *inpb.InvocationEvent) error {
+	inv, err := LookupInvocationWithCallback(ctx, env, iid, maxEventCount, func(event *inpb.InvocationEvent) error {
```

Comment:
a possibly nicer / more idiomatic way to do this would be to have your callback return a bool indicating if you want to continue. Then track maxEventCount in your callback